Cod3rsGrowth.Servicos/Servicos/ServicoCarro.cs

```diff
@@ -24,9 +28,24 @@ public Carro ObterPorId(int IdDoItem)
 
         public void Criar(Carro carro)
         {
+            var resultado = _validadorCarro.Validate(carro);
+            var erros = "";
+            if (!resultado.IsValid)
+            {
+                foreach (var falhas in resultado.Errors)
+                {
+                    erros += falhas.ErrorMessage + " ";
+                }
+                throw new ValidationException(erros);
+            }
             _repositorioCarro.Criar(carro);
         }
+        public int ObterNovoId()
```

Comment:
Vai ter que remover essa logica de ObterNovoId do serviço. Esse id não deve ser um tipo de serviço, ele tem que ser uma implementação interna do seu repositório. Deve acontecer dentro do Criar do repositório, no momento em que for atribuir um novo item a lista, ele deve manipular o objeto atribuir um id e então depois disso ele é add na lista.